src/components/common/Image/Image.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useState } from "react";
+import clsx from "clsx";
+import defaultPlaceholderUrl from "url:/assets/placeholder.png";
+import { sleep } from "~utils/promises/sleep";
+import { useTheme } from "~components/embed/contexts/ThemeContext";
+
+import styles from "./Image.module.scss";
+
+const fakeDelay = process.env.NODE_ENV === "development" ? 2500 : 0;
+
+type BorderRadiusVariant = "none" | "rounded" | "circular";
+
+interface ImageProps {
+  src: string;
+  srcDark?: string;
+  width: number;
+  height: number;
+  title?: string;
+  alt?: string;
+  className?: string;
+  fullWidth?: boolean;
+  border?: boolean;
+  borderRadius?: BorderRadiusVariant;
+  backgroundColor?: string;
+  placeholderURL?: string;
+  pointer?: [number, number];
+  style?: React.CSSProperties;
+}
+
+export default function Image({
+  src,
+  srcDark,
+  width,
+  height,
+  title,
+  alt = "",
+  className,
+  fullWidth,
+  border,
+  borderRadius = "none",
+  backgroundColor = "transparent",
+  placeholderURL = defaultPlaceholderUrl,
+  pointer: pointerProp,
+  style,
+}: ImageProps) {
+  const theme = useTheme();
+  const displayTheme = theme.mode;
+
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const pictureClassName = clsx(className, styles.root, {
+    [styles.fullWidth]: fullWidth,
+    [styles.hasBorder]: border,
+    [styles.isRounded]: borderRadius === "rounded",
+    [styles.isCircular]: borderRadius === "circular",
+    [styles.isLoaded]: isLoaded,
+  });
+
+  const pictureStyle = {
+    "--imgWidth": `${width}px`,
+    "--imgHeight": fullWidth ? "auto" : `${height}px`,
+    "--imgAspectRatio": fullWidth ? width / height : "auto",
+    "--imgBackgroundColor": backgroundColor,
+    "--imgPlaceholderBackgroundURL": placeholderURL && placeholderURL !== "none" ? `url(${placeholderURL})` : "none",
+    ...style,
+  };
+
+  const imgClassName = clsx(styles.img, {
+    [styles.isLoaded]: isLoaded,
+  });
+
+  const handleImageLoaded = async () => {
+    if (fakeDelay) await sleep(fakeDelay + Math.random() * fakeDelay * 2);
+
+    setIsLoaded(true);
+  };
+
+  const [devPointer, setDevPointer] = useState<[number, number] | null>(null);
+
+  const pointer = devPointer || pointerProp;
+
+  const movePointer =
```

Comment:
In development, clicking on an image will show a "click indicator" on it and log its position in the console for you to copy it and add it to the code.